### PR TITLE
Warm start errors

### PR DESCRIPTION
**Pull request by inventormc**
Explicitly error when users try to set parameters that can't be set with early stopping (`n_estimators` for ensemble warm starting and `max_iter` for iteration warm starting).

### CODE REVIEW

File: tune_sklearn/utils.py [hunk 1]
```diff
@@ -1,6 +1,16 @@
 from sklearn.metrics import check_scoring
 from sklearn.pipeline import Pipeline
+from tune_sklearn._detect_xgboost import is_xgboost_model
 import numpy as np
+from enum import Enum
+
+
+class EarlyStopping(Enum):
```
Comment (richardliaw) [inline tune_sklearn/utils.py:8]: docstring?

File: tune_sklearn/utils.py [hunk 2]
```diff
@@ -1,6 +1,16 @@
...
+    PARTIAL_FIT = 1
+    WARM_START_ITER = 2
+    WARM_START_ENSEMBLE = 3
+    XGB = 4
+    NO_EARLY_STOP = 5
```
Comment (richardliaw) [inline tune_sklearn/utils.py:13]: prefer string value instead of numeric (for better readability when serialized/debugged)

**Review by richardliaw — APPROVED** [review]
wow this is clean! left a couple comments for the treatment of the enum, but feel free to merge when addressed!

**Comment by richardliaw** [comment]
What's the diff between this and #105 ?

**Comment by inventormc** [comment]
> What's the diff between this and #105 ?

This was supposed to only handle the errors for warm start, but I had to take commits from #105 since this relies on the enum changes in that pr.

**Pull request merged**